openprescribing/dmd/tests.py

```diff
@@ -139,11 +127,41 @@ def test_import_dmd(self):
             'Diclofenac diethylammonium'
         )
 
-        # This checks that records with the same primary key can be imported
-        # multiple times.
+        # Now create a TariffPrice object referencing a product whose VPID will
+        # change in the next import.
+
+        vmpp = DMDVmpp.objects.get(pk=22479411000001100)
+        product = DMDProduct.objects.get(dmdid=vmpp.vpid)
+
+        tp = TariffPrice.objects.create(
+            date='2018-07-01',
+            vmpp=vmpp,
+            product=product,
+            tariff_category=product.tariff_category,
+            price_pence=100,
+        )
+
+        # Import another data dump.  This data is identical to that in dmd/1,
+        # except that the VMP with VPID 22480211000001104 has been updated with
+        # a new VPID (12345).  This VMP now has a VPIDPREV field, and all
+        # references to the old VPID have been updated to the new VPID.
+
+        path = 'dmd/tests/fixtures/dmd/2/dmd.zip'
         with patch('zipfile.ZipFile'):
             call_command('import_dmd', '--zip_path', path)
 
+        # Check that there is no DMDProduct or VMP with the old VPID.
+        self.assertFalse(DMDProduct.objects.filter(dmdid=22480211000001104).exists())
+        self.assertQuery(
+            'SELECT count(*) FROM dmd_vmp WHERE vpid = 22480211000001104',
+            0
+        )
+
+        # Check that the TariffPrice has been updated to reference a new
+        # DMDProduct.
+        tp.refresh_from_db()
+        self.assertEqual(tp.product_id, 12345)
```

Comment:
Maybe check the VMPP foreign key here, too, particularly given we plan to excise DMDProduct in the nearish future?

Reply:
`DMDVmpp` doesn't have a foreigin key to `DMDProduct`.

Reply:
I mean `self.assertEqual(tp.vmpp_id, 12345)`

Reply:
I don't think you do mean that!  But I've added a test at https://github.com/ebmdatalab/openprescribing/blob/96ca0642373273f990b42fdf08acc93a4fff470d/openprescribing/dmd/tests.py#L165-L171 which should cover what I think you mean.